internal/bridge/crunchybridgecluster/crunchybridgecluster_controller.go

```diff
@@ -328,9 +330,53 @@ func (r *CrunchyBridgeClusterReconciler) Reconcile(ctx context.Context, req ctrl
 
 		for _, cluster := range clusters {
 			if crunchybridgecluster.Name == cluster.Name {
```

Comment:
❓ Should this be `crunchybridgecluster.Spec.ClusterName`?

Reply:
```suggestion
			if crunchybridgecluster.Spec.ClusterName == cluster.Name {
```
Yeah, this code was in place before we separated out the name of the CR from the name of the cluster.

I think we separated them out in order to make the distinction between the CR requirements and the bridge requirements for names. (Also there's differing constraints around uniqueness in k8s / Bridge.)

Reply:
I'm partial to the idea of just having one name and enforcing constraints for both bridge and k8s when creating the CR... Only scenario that I can come up with where this is problematic is if the customer is adopting a pre-existing cluster in bridge that has a name that conflicts with certain kubernetes constraints, but if that's the case I guess we just make the customer change the name of the cluster in bridge?

Reply:
The K8s name and Bridge name are different ideas with different constraints, so we need separate fields.

We might be able to enforce some similarities or default-ish behavior using cross-field CEL validation in the future. In that case, the `spec.clusterName` field could become optional when the K8s name is also a valid Bridge name.

Reply:
🤔 Notably, K8s names are immutable and Bridge names are mutable.